### PR TITLE
[Linting] fix outstanding linter errors

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Fix linter errors exposed in #674

## Issue

- #674

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: pkg/appgateserver/server.go [hunk 1]
```diff
@@ -300,13 +300,13 @@ func (app *appGateServer) ServePprof(ctx context.Context, addr string) error {
 	// If no error, start the server in a new goroutine
 	go func() {
 		app.logger.Info().Str("endpoint", addr).Msg("starting a pprof endpoint")
-		server.ListenAndServe()
+		_ = server.ListenAndServe()
```
Comment (bryanchriswhite) [inline pkg/appgateserver/server.go:303]: Should we do something here if there's an error?

File: pkg/relayer/relayminer.go [hunk 2]
```diff
@@ -121,13 +121,13 @@ func (rel *relayMiner) ServePprof(ctx context.Context, addr string) error {
 	// If no error, start the server in a new goroutine
 	go func() {
 		rel.logger.Info().Str("endpoint", addr).Msg("starting a pprof endpoint")
-		server.ListenAndServe()
+		_ = server.ListenAndServe()
```
Comment (bryanchriswhite) [inline pkg/relayer/relayminer.go:124]: Should we do something here if there's an error?

File: testutil/testproxy/relayerproxy.go [hunk 3]
```diff
@@ -151,10 +151,14 @@ func WithServicesConfigMap(
 				server.Handler = http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
 					sendJSONRPCResponse(test.t, w)
 				})
-				go func() { server.ListenAndServe() }()
+				go func() {
+					err := server.ListenAndServe()
+					require.NoError(test.t, err)
```
Comment (bryanchriswhite) [inline testutil/testproxy/relayerproxy.go:156]: This seems to be producing new `bind: address already in use` errors :thinking: :

image

File: pkg/appgateserver/server.go (hunk 1 above)
Comment (okdas) [reply pkg/appgateserver/server.go:303]: Let's add an error log here. That way, the user/operator can know why the endpoint is not available if something goes wrong (such as a port already in use).

File: pkg/relayer/relayminer.go (hunk 2 above)
Comment (okdas) [reply pkg/relayer/relayminer.go:124]: Let's log the error.

File: testutil/testproxy/relayerproxy.go (hunk 3 above)
Comment (okdas) [reply testutil/testproxy/relayerproxy.go:156]: There were a couple of issues:
- We tried to create new proxies with the same config while already having one running for the duration of whole test.
- We did not cleanup the servers properly. 

The fix incoming! :)

**Review by okdas — APPROVED** [review]
Thank you so much for taking an initiative on this effort! 
LGTM, and test seem to pass now.

If you're good with the changes I added - let's merge this in.

**Pull request merged**